src/components/FileUpload/FileUpload.tsx

```diff
@@ -0,0 +1,475 @@
+import React, { useEffect } from "react";
+import styled, { css } from "styled-components";
+import { useState, useRef, useCallback } from "react";
+import { Text } from "@/components/Typography/Text/Text";
+import { Button, Icon, IconButton, Title } from "@/components";
+
+interface FileInfo {
+  name: string;
+  size: number;
+}
+
+interface FileUploadProps {
+  title: string;
+  supportedFileTypes?: string[];
+  size?: "sm" | "md";
+  progress?: number;
+  showSuccess?: boolean;
+  showProgress?: boolean;
+  failureMessage?: string;
+  onRetry?: () => void;
+  onFileSelect?: (file: File) => void;
+  onFileSelectFailure?: () => void;
+}
+
+const UploadArea = styled.div<{
+  $isDragging: boolean;
+  $size: "sm" | "md";
+  $hasFile: boolean;
+  $isError?: boolean;
+}>`
+  background-color: ${({ theme }) => theme.click.fileUpload.color.background.default};
+  border: ${({ theme }) => `1px solid ${theme.click.fileUpload.color.stroke.default}`};
+  border-radius: ${({ theme }) => `${theme.click.fileUpload.md.radii.all}`};
+  padding: ${props => (!props.$hasFile ? "16px" : "8px")};
+  display: flex;
+  flex-direction: ${props =>
+    props.$hasFile ? "row" : props.$size === "sm" ? "row" : "column"};
+  align-items: center;
+  justify-content: ${props =>
+    props.$hasFile ? "space-between" : props.$size === "sm" ? "space-between" : "center"};
+  gap: 12px;
+  cursor: ${props => (props.$hasFile ? "default" : "pointer")};
+  transition: all 0.2s ease;
+
+  ${props =>
+    !props.$hasFile &&
+    css`
+      border-style: dashed;
+      border-color: ${({ theme }) => theme.click.card.primary.color.stroke.default};
+
+      &:focus {
+        background-color: ${({ theme }) =>
+          theme.click.card.secondary.color.background.hover};
+
+        button {
+          background-color: ${({ theme }) =>
+            theme.click.card.secondary.color.background.hover};
+        }
+      }
+
+      ${props.$isDragging &&
+      css`
+        background-color: ${({ theme }) =>
+          theme.click.card.secondary.color.background.hover};
+        border-color: #9ca3af;
+      `}
+    `}
+
+  ${props =>
+    props.$isError &&
+    css`
+      background-color: ${({ theme }) => theme.click.fileUpload.color.background.error};
+      border: none;
+    `}
+`;
+
+const FileUploadTitle = styled(Title)`
+  font-family: ${({ theme }) => theme.click.fileUpload.typography.title.default};
+  color: ${({ theme }) => theme.click.fileUpload.color.title.default};
+`;
+
+const FileUploadDescription = styled(Text)`
+  font-family: ${({ theme }) => theme.click.fileUpload.typography.description.default};
+  color: ${({ theme }) => theme.click.fileUpload.color.description.default};
+`;
+
+const DocumentIcon = styled(Icon)`
```

Comment:
This naming could potentially constrain us if we wish to make the icon configurable, ie. upload an image or something? Is there anything else we could use to differentiate?